### PR TITLE
Completed issue-3-edit-entry. - Davin Phuong

### PR DESCRIPTION
**Pull request by dn-phuong**
Kapture 2022-03-02 at 15 27 16

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -13,32 +13,46 @@ var $form = document.querySelector('form');
...
+    var entryInputs = {
+      title: title,
+      photoUrl: photoUrl,
+      notes: notes,
+      entryId: data.nextEntryId
+    };
+    data.nextEntryId++;
+    data.entries.unshift(entryInputs);
+    $list.prepend(renderEntry(entryInputs));
+    data.editing = null;
```
Comment (codingcodymiller) [inline js/main.js:30]: You can remove this line, you already have it below the conditional, so it will always happen.

File: js/main.js [hunk 2]
```diff
@@ -13,32 +13,46 @@ var $form = document.querySelector('form');
...
+    data.editing.notes = notes;
+
+    for (var i = 0; i < data.editing.length; i++) {
+      var $listItems = document.querySelectorAll('li');
+      var editEntryId = $listItems[i].getAttribute('data-entry-id');
+      editEntryId = parseInt(editEntryId);
+
+      if (data.editing[i].entryId === editEntryId) {
+        data.editing[i] = editEntryId;
+      }
```
Comment (codingcodymiller) [inline js/main.js:44]: What is the purpose of this?

File: js/main.js [hunk 3]
```diff
@@ -113,3 +131,28 @@ if (data.view === 'entries') {
...
+  var header = document.querySelector('h2');
+  header.textContent = 'Edit Entry';
+  if (event.target && event.target.matches('i')) {
+    $entryForm.className = '';
+    $entries.className = 'hidden';
+    data.view = 'entry-form';
+
+    $closestListItem = event.target.closest('li');
+    var currentId = $closestListItem.getAttribute('data-entry-id');
+    currentId = JSON.parse(currentId);
```
Comment (codingcodymiller) [inline js/main.js:147]: Is this to convert it to a number? You should just use `parseInt`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by dn-phuong** [comment]
new test:
Kapture 2022-03-02 at 16 41 00

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**